kolibri/plugins/coach/assets/src/modules/questionDetail/handlers.js

```diff
@@ -57,7 +62,9 @@ function showQuestionDetailView(params) {
         exercise.assessmentmetadata = exercise.assessmentmetadata || {};
         let title;
         if (exam) {
-          const question = exam.question_sources.find(source => source.item === questionId);
+          const question = exam.question_sources[0].questions.reduce((acc, source) => {
```

Comment:
This seems like it doesn't need a reducer.

Basically, I think, this will basically always return the last item in the `questions` list.

For example:

```
[{ thing: 1 }, { thing: 2 }].reduce((acc, source) => {
     return source;  // This sets acc to the value returned
 } , []);

// Iteration 1:
// acc = []  // acc is set to the value of the second arg passed to `reduce`
// source = { thing: 1 }
// reducer callback returns: { thing: 1 }
// note: the value of acc is no longer an array because we returned `source` _aka_ {  thing: 1 } 

// Iteration 2:
// acc = { thing: 1 }   // The value returned by the last iteration
// source = { thing: 2 }
// reducer callback returns: { thing: 2 }

// Finally:
// acc = { thing: 2 }  // The value returned by the last iteration
// the expression altogether returns { thing: 2 }
```

That's how I try to think about reducers, anyhow. It's mainly important to remember that what you return in the callback function is the value of `acc` on the next iteration and the final value of `acc` is the final return value.

Reply:
So following up here, I think that considering @rtibbles thoughts on the code above, you can reduce the sections kind of like how it is done [here](https://github.com/nucleogenesis/kolibri/blob/5aa75bf7f202e6019be1114254a1c0e72a7d092a/kolibri/core/assets/src/exams/utils.js#L178)